libs/langchain/langchain/llms/ollama.py

```diff
@@ -0,0 +1,209 @@
+import json
+from typing import Any, Dict, Iterator, List, Mapping, Optional
+
+import requests
+from pydantic import Extra
+
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+from langchain.llms.base import LLM
+from langchain.schema.language_model import BaseLanguageModel
+from langchain.schema.output import GenerationChunk
+
+
+def _stream_response_to_generation_chunk(
+    stream_response: str,
+) -> GenerationChunk:
+    """Convert a stream response to a generation chunk."""
+    return GenerationChunk(
+        text=json.loads(stream_response.strip()).get("response", ""),
+    )
+
+
+class _OllamaCommon(BaseLanguageModel):
```

Comment:
don't think we need to split out _OllamaCommon if there's only one child class, unless we're planning on adding a ChatOllama soon?

Reply:
Yeah I figured it was a possibility, can recombine them though

Reply:
Let's leave it, no point in recombining actually and the potential is there